xc/device.ml

```diff
@@ -1779,6 +1881,9 @@ module Dm = struct
       if not !finished then
         raise (Ioemu_failed (name, "Timeout reached while starting daemon"))
     end;
+
+    if is_upstream_qemu domid then
+      QMP_Event.wakeup domid;
```

Comment:
This should be:
```
QMP_Event.add domid;
```

Also, this chunk implements:
```
623 After a QEMU instance is created, the toolstack adds (QMP_event_fd, domid)
624 to a global toolstack table, so that when handling a QMP event connection from
625 an fd, the thread is able find the corresponding domid and to call a function
626 `qmp_event_handle event domid`.
```
but what about this bit?
```
626                                  When a QEMU instance is destroyed or crashes,
627 the toolstack should remove (QMP_event_fd, domid) from this table to prevent
628 a memory leak.
```

Reply:
When a QEMU instance is destroyed,  epoll will get `EPOLLHUP` event, then the code will remove (QMP_event_fd, domid) from this table.

It's better to remove it when we know qemu will be destroyed , now add `QMP_Event.remove domid;` when stop qemu.
```
     let stop_qemu () = (match (Qemu.pid ~xs domid) with
         | None -> () (* nothing to do *)
         | Some qemu_pid ->
+          if is_upstream_qemu domid then
+            QMP_Event.remove domid;
```